### DISCUSSION TITLE
Micronaut Data JDBC with Database-per-tenant Model

### DISCUSSION THREAD

**Discussion by shah-dhwanil** [micronaut-projects/micronaut-data#1442] [Q&A]
Hello!
I would like to request some guidance on how to create repository as I have a database per tenant model. So I have to determine the data source to be used at runtime. So how to implement this with Micronaut Data JDBC.

**Comment by dstepanov** [comment]
You can provide a custom `BeanCreatedEventListener<DataSource>` which will use thread-local to delegate to the chosen data source.
